src/compare-tree/key-path.js

```diff
@@ -0,0 +1,40 @@
+const KEY_DELIMETER  = '.'
+const PATH_DELIMETER = ','
```

Comment:
open to some flexibility on this? I would expect something like
```
const KEY_DELIMETER  = '-'
const PATH_DELIMETER = ','
```
if i'm understanding what these two do, i don't actually see anything splitting on the `KEY_DELIMETER` which i'm confused by